### DISCUSSION TITLE
updateAsDraft returns boolean

### DISCUSSION THREAD

**Discussion by SergkeiM** [oddvalue/laravel-drafts#21] [Ideas]
Currently updateAsDraft returns boolean,

So when you update a `Model` as Draft there is no possibilty to attach relationship to new `Draft Model`

Example:

```php
$draft = $page->updateAsDraft($request->all());

$draft->domains()->attach([1, 2, 3]); //fails as draft is not new `Model` created but `boolean`
```

Note: will create a PR if this sounds legit to implement

**Comment by oddvalue** [comment]
@Froxz Yeah, I can see how that might be confusing. It's set up that way to replicate the functionality of the `update` method on the base model. While I can see the value in having it return the draft instead that would be a breaking change in the package. Perhaps there could be a different method that returns the draft after saving.
